File: AlgoPersonalStudy/src/backjoon/gold/BJ_2749.java
package backjoon.gold;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class BJ_2749 { // not solved alone
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long n = Long.parseLong(br.readLine());
        int divNum = 1000000;

        int p = divNum / 10 * 15;
        int[] fibo = new int[p];
        
        for(int i=2;i<p;i++){
            fibo[i] = fibo[(i-1)] + fibo[(i-2)];
            fibo[i] %= divNum;
        }
        System.out.println(fibo[(p-1)]);
    }
}

File: AlgoPersonalStudy/src/_kotlin/backjoon/gold/BJ_2749.kt
package _kotlin.backjoon.gold

import java.io.BufferedReader
import java.io.InputStreamReader

class BJ_2749 {
    fun solution(){
        val br = BufferedReader(InputStreamReader(System.`in`))
        val n = br.readLine().toLong()
        val divNum = 1000000

        val p = divNum / 10 * 15
        val fibo = IntArray(p)

        for (i in 2 until p) {
            fibo[i] = fibo[i - 1] + fibo[i - 2]
            fibo[i] %= divNum
        }
        println(fibo[p - 1])
    }
}
fun main(){

}